parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -87,35 +96,35 @@ def file_iter(self, filename, start=0):
                 yield line_number, line
         yield line_number, ""  # to end the last entry if not ended
 
-    def normalizing(self, line, lang="default"):
-        """normalize a string depending of the language code lang"""
+    def normalizing(self, line, lang="default", char="-"):
+        """Normalize a string depending on the language code"""
         line = unicodedata.normalize("NFC", line)
 
-        # removing accent
+        # Removing accent
         if lang in ["fr", "ca", "es", "it", "nl", "pt", "sk", "en"]:
-            line = re.sub(r"[¢£¤¥§©ª®°²³µ¶¹º¼½¾×‰€™]", "-", line)
+            line = re.sub(r"[¢£¤¥§©ª®°²³µ¶¹º¼½¾×‰€™]", char, line)
             line = unidecode.unidecode(line)
 
-        # lower case except if language in list
+        # Lower case except if language in list
         if lang not in []:
             line = line.lower()
 
-        # changing unwanted character to "-"
-        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
-        line = re.sub(r"&\w+;", "-", line)
+        # Changing unwanted character to "-"
+        line = re.sub(r"[\u0000-\u0027\u200b]", char, line)
+        line = re.sub(r"&\w+;", char, line)
         line = re.sub(
             r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]",  # noqa: E501
-            "-",
+            char,
             line,
         )
 
-        # removing excess "-"
-        line = re.sub(r"-+", "-", line)
-        line = line.strip("-")
+        # Removing excess "-"
+        line = re.sub(r"-+", char, line)
+        line = line.strip(char)
         return line
```

Comment:
I would like to avoid code duplication.

Backend must be able to use parser has a library (you can (add `-e ../parser/openfoodfacts_taxonomy_parser` in requirements.txt, and in the Docker we have to copy the parser path in the docker)).

So:
1. here we should put normalizing in a separate function (in normalizer.py is even better)
2. call this function in this method
3. import from parser and call it in backend instead of having a copy of normalizer.py there.

It's ok if you want to do it on next step (creating a project), but this really has to be done. Code duplication means possible divergence, which is bad !

Reply:
@alexgarel I changed up all the imports and called the functions appropriately. I'm facing some issues with changing the docker related files. Would you be able to help me out?